code/game/machinery/doors/airlock.dm

```diff
@@ -416,6 +416,8 @@
 				if(G.siemens_coefficient)//not insulated
 					new /datum/hallucination/shock(H)
 					return
+	if(SEND_SIGNAL(src, COMSIG_AIRLOCK_TOUCHED, user) & COMPONENT_PREVENT_OPEN)
+		return
```

Comment:
This won't send anything if the target got zapped, is this intentional?

Reply:
Yeah that way it works similarly to normal airlocks